<!-- markdownlint-disable -->
<h1 align="center">
    best-of-NGS-Germline
    <br>
</h1>

## Contents

- [FASTQ quality control](#fastq-quality-control) _1 projects_
- [Read alignment](#read-alignment) _2 projects_
- [BAM manipulation and stats](#bam-manipulation-and-stats) _4 projects_
- [Variant calling](#variant-calling) _3 projects_
- [VCF manipulation and stats](#vcf-manipulation-and-stats) _1 projects_
- [Copy number variation calling](#copy-number-variation-calling) _1 projects_
- [Structural variation calling](#structural-variation-calling) _1 projects_
- [Variant annotation](#variant-annotation) _3 projects_
- [Gene typing](#gene-typing) _4 projects_
- [BED manipulation and stats](#bed-manipulation-and-stats) _3 projects_
- [Relationship inference](#relationship-inference) _1 projects_
- [FASTQ compression](#fastq-compression) _1 projects_

<br>

## FASTQ quality control

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/OpenGene/fastp">OpenGene/fastp</a></b> (🥇20 ·  ⭐ 1.8K) - An ultra-fast all-in-one FASTQ preprocessor (QC/adapters/trimming/filtering/splitting/merging...).</summary>

- [GitHub](https://github.com/OpenGene/fastp) (👨‍💻 27 · 🔀 330 · 📋 500 - 64% open · ⏱️ 07.04.2024):

	```
	git clone https://github.com/OpenGene/fastp
	```
</details>
<br>

## Read alignment

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/lh3/bwa">lh3/bwa</a></b> (🥇21 ·  ⭐ 1.5K) - Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment).</summary>

- [GitHub](https://github.com/lh3/bwa) (👨‍💻 33 · 🔀 510 · 📥 130K · 📋 290 - 64% open · ⏱️ 15.04.2024):

	```
	git clone https://github.com/lh3/bwa
	```
</details>
<details><summary><b><a href="https://github.com/bwa-mem2/bwa-mem2">bwa-mem2/bwa-mem2</a></b> (🥉15 ·  ⭐ 680) - The next version of bwa-mem.</summary>

- [GitHub](https://github.com/bwa-mem2/bwa-mem2) (👨‍💻 12 · 🔀 87 · 📥 29K · 📋 200 - 38% open · ⏱️ 18.12.2023):

	```
	git clone https://github.com/bwa-mem2/bwa-mem2
	```
</details>
<br>

## BAM manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/samtools">samtools/samtools</a></b> (🥇24 ·  ⭐ 1.5K · 📉) - Tools (written in C using htslib) for manipulating next-generation sequencing data.</summary>

- [GitHub](https://github.com/samtools/samtools) (👨‍💻 94 · 🔀 550 · 📥 1.5M · 📋 1.2K - 13% open · ⏱️ 18.04.2024):

	```
	git clone https://github.com/samtools/samtools
	```
</details>
<details><summary><b><a href="https://github.com/brentp/mosdepth">brentp/mosdepth</a></b> (🥈21 ·  ⭐ 650 · 📈) - fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing.</summary>

- [GitHub](https://github.com/brentp/mosdepth) (👨‍💻 17 · 🔀 100 · 📥 320K · 📋 200 - 26% open · ⏱️ 17.04.2024):

	```
	git clone https://github.com/brentp/mosdepth
	```
</details>
<details><summary><b><a href="https://github.com/biod/sambamba">biod/sambamba</a></b> (🥉20 ·  ⭐ 550 · 💤) - Tools for working with SAM/BAM data.</summary>

- [GitHub](https://github.com/biod/sambamba) (👨‍💻 29 · 🔀 98 · 📥 140K · 📋 450 - 6% open · ⏱️ 21.08.2023):

	```
	git clone https://github.com/biod/sambamba
	```
</details>
<details><summary><b><a href="https://github.com/GregoryFaust/samblaster">GregoryFaust/samblaster</a></b> (🥉14 ·  ⭐ 210 · 💀) - samblaster: a tool to mark duplicates and extract discordant and split reads from sam files.</summary>

- [GitHub](https://github.com/GregoryFaust/samblaster) (👨‍💻 5 · 🔀 28 · 📥 45K · 📋 49 - 26% open · ⏱️ 04.06.2020):

	```
	git clone https://github.com/GregoryFaust/samblaster
	```
</details>
<br>

## Variant calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/google/deepvariant">google/deepvariant</a></b> (🥇25 ·  ⭐ 3.1K) - DeepVariant is an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA..</summary>

- [GitHub](https://github.com/google/deepvariant) (👨‍💻 30 · 🔀 680 · 📥 4.7K · 📋 750 - 0% open · ⏱️ 18.03.2024):

	```
	git clone https://github.com/google/deepvariant
	```
</details>
<details><summary><b><a href="https://github.com/broadinstitute/gatk">broadinstitute/gatk</a></b> (🥇25 ·  ⭐ 1.6K) - Official code repository for GATK versions 4 and up.</summary>

- [GitHub](https://github.com/broadinstitute/gatk) (👨‍💻 130 · 🔀 560 · 📥 600K · 📋 4.5K - 26% open · ⏱️ 18.04.2024):

	```
	git clone https://github.com/broadinstitute/gatk
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/strelka">Illumina/strelka</a></b> (🥉18 ·  ⭐ 350 · 💀) - Strelka2 germline and somatic small variant caller.</summary>

- [GitHub](https://github.com/Illumina/strelka) (👨‍💻 32 · 🔀 100 · 📥 35K · 📋 240 - 53% open · ⏱️ 06.02.2019):

	```
	git clone https://github.com/Illumina/strelka
	```
</details>
<br>

## VCF manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/bcftools">samtools/bcftools</a></b> (🥇24 ·  ⭐ 620) - This is the official development repository for BCFtools. See installation instructions and other documentation here..</summary>

- [GitHub](https://github.com/samtools/bcftools) (👨‍💻 54 · 🔀 230 · 📥 600K · 📋 1.8K - 16% open · ⏱️ 19.04.2024):

	```
	git clone https://github.com/samtools/bcftools
	```
</details>
<br>

## Copy number variation calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/etal/cnvkit">etal/cnvkit</a></b> (🥇21 ·  ⭐ 470 · 📈) - Copy number variant detection from targeted DNA sequencing.</summary>

- [GitHub](https://github.com/etal/cnvkit) (👨‍💻 37 · 🔀 150 · 📦 28 · 📋 730 - 39% open · ⏱️ 12.04.2024):

	```
	git clone https://github.com/etal/cnvkit
	```
</details>
<br>

## Structural variation calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/Illumina/manta">Illumina/manta</a></b> (🥇20 ·  ⭐ 390 · 💀) - Structural variant and indel caller for mapped sequencing data.</summary>

- [GitHub](https://github.com/Illumina/manta) (👨‍💻 24 · 🔀 150 · 📥 43K · 📋 300 - 42% open · ⏱️ 17.07.2019):

	```
	git clone https://github.com/Illumina/manta
	```
</details>
<br>

## Variant annotation

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/Ensembl/ensembl-vep">Ensembl/ensembl-vep</a></b> (🥇19 ·  ⭐ 430) - The Ensembl Variant Effect Predictor predicts the functional effects of genomic variants.</summary>

- [GitHub](https://github.com/Ensembl/ensembl-vep) (👨‍💻 45 · 🔀 150 · 📋 1K - 10% open · ⏱️ 17.01.2024):

	```
	git clone https://github.com/Ensembl/ensembl-vep
	```
</details>
<details><summary><b><a href="https://github.com/pcingola/SnpEff">pcingola/SnpEff</a></b> (🥉14 ·  ⭐ 230) - </summary>

- [GitHub](https://github.com/pcingola/SnpEff) (👨‍💻 21 · 🔀 70 · 📋 480 - 6% open · ⏱️ 09.04.2024):

	```
	git clone https://github.com/pcingola/SnpEff
	```
</details>
<details><summary><b><a href="https://github.com/WGLab/doc-ANNOVAR">WGLab/doc-ANNOVAR</a></b> (🥉13 ·  ⭐ 210 · 📈) - Documentation for the ANNOVAR software.</summary>

- [GitHub](https://github.com/WGLab/doc-ANNOVAR) (👨‍💻 8 · 🔀 310 · 📋 230 - 47% open · ⏱️ 09.04.2024):

	```
	git clone https://github.com/WGLab/doc-ANNOVAR
	```
</details>
<br>

## Gene typing

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/FRED-2/OptiType">FRED-2/OptiType</a></b> (🥇14 ·  ⭐ 170 · 💀) - Precision HLA typing from next-generation sequencing data.</summary>

- [GitHub](https://github.com/FRED-2/OptiType) (👨‍💻 13 · 🔀 70 · 📋 110 - 41% open · ⏱️ 15.09.2020):

	```
	git clone https://github.com/FRED-2/OptiType
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/ExpansionHunter">Illumina/ExpansionHunter</a></b> (🥈13 ·  ⭐ 170 · 📉) - A tool for estimating repeat sizes.</summary>

- [GitHub](https://github.com/Illumina/ExpansionHunter) (👨‍💻 17 · 🔀 52 · 📥 10K · 📋 110 - 60% open · ⏱️ 30.01.2024):

	```
	git clone https://github.com/Illumina/ExpansionHunter
	```
</details>
<details><summary><b><a href="https://github.com/humanlongevity/HLA">humanlongevity/HLA</a></b> (🥉8 ·  ⭐ 94 · 💀) - xHLA: Fast and accurate HLA typing from short read sequence data.</summary>

- [GitHub](https://github.com/humanlongevity/HLA) (👨‍💻 4 · 🔀 46 · 📋 63 - 73% open · ⏱️ 04.10.2017):

	```
	git clone https://github.com/humanlongevity/HLA
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/SMNCopyNumberCaller">Illumina/SMNCopyNumberCaller</a></b> (🥉8 ·  ⭐ 48) - A copy number caller for SMN1 and SMN2 to enable SMA diagnosis and carrier screening with WGS.</summary>

- [GitHub](https://github.com/Illumina/SMNCopyNumberCaller) (👨‍💻 4 · 🔀 11 · 📋 10 - 50% open · ⏱️ 14.10.2023):

	```
	git clone https://github.com/Illumina/SMNCopyNumberCaller
	```
</details>
<br>

## BED manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/arq5x/bedtools2">arq5x/bedtools2</a></b> (🥇24 ·  ⭐ 900 · 📈) - bedtools - the swiss army knife for genome arithmetic.</summary>

- [GitHub](https://github.com/arq5x/bedtools2) (👨‍💻 94 · 🔀 280 · 📥 1M · 📋 820 - 26% open · ⏱️ 14.04.2024):

	```
	git clone https://github.com/arq5x/bedtools2
	```
</details>
<details><summary><b><a href="https://github.com/lh3/bedtk">lh3/bedtk</a></b> (🥉7 ·  ⭐ 120 · 💀) - A simple toolset for BED files (warning: CLI may change before bedtk becomes stable).</summary>

- [GitHub](https://github.com/lh3/bedtk) (👨‍💻 3 · 🔀 13 · ⏱️ 01.04.2022):

	```
	git clone https://github.com/lh3/bedtk
	```
</details>
<details><summary><b><a href="https://github.com/lh3/tabtk">lh3/tabtk</a></b> (🥉4 ·  ⭐ 59 · 💀) - Toolkit for processing TAB-delimited format.</summary>

- [GitHub](https://github.com/lh3/tabtk) (🔀 11 · 📋 2 - 50% open · ⏱️ 11.08.2016):

	```
	git clone https://github.com/lh3/tabtk
	```
</details>
<br>

## Relationship inference

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/brentp/somalier">brentp/somalier</a></b> (🥇16 ·  ⭐ 250) - fast sample-swap and relatedness checks on BAMs/CRAMs/VCFs/GVCFs... like damn that is one smart wine guy.</summary>

- [GitHub](https://github.com/brentp/somalier) (👨‍💻 8 · 🔀 33 · 📥 14K · 📋 120 - 39% open · ⏱️ 02.02.2024):

	```
	git clone https://github.com/brentp/somalier
	```
</details>
<br>

## FASTQ compression

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/shubhamchandak94/Spring">shubhamchandak94/Spring</a></b> (🥇10 ·  ⭐ 110 · 💤) - FASTQ compression.</summary>

- [GitHub](https://github.com/shubhamchandak94/Spring) (🔀 21 · ⏱️ 27.05.2023):

	```
	git clone https://github.com/shubhamchandak94/Spring
	```
</details>
