## Contents

- [FASTQ quality control](#fastq-quality-control) _1 projects_
- [Read alignment](#read-alignment) _1 projects_
- [BAM manipulation and stats](#bam-manipulation-and-stats) _4 projects_
- [Variant calling](#variant-calling) _0 projects_

<br>

## FASTQ quality control

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/OpenGene/fastp">fastp</a></b> (ğŸ¥‡19 Â·  â­ 1.7K Â· ğŸ’¤) - An ultra-fast all-in-one FASTQ preprocessor (QC/adapters/trimming/filtering/splitting/merging...).</summary>

- [GitHub](https://github.com/OpenGene/fastp) (ğŸ‘¨â€ğŸ’» 26 Â· ğŸ”€ 310 Â· ğŸ“‹ 490 - 64% open Â· â±ï¸ 31.05.2023):

	```
	git clone https://github.com/OpenGene/fastp
	```
</details>
<br>

## Read alignment

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/lh3/bwa">bwa</a></b> (ğŸ¥‡19 Â·  â­ 1.4K Â· ğŸ’€) - Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment).</summary>

- [GitHub](https://github.com/lh3/bwa) (ğŸ‘¨â€ğŸ’» 33 Â· ğŸ”€ 510 Â· ğŸ“¥ 100K Â· ğŸ“‹ 280 - 63% open Â· â±ï¸ 22.09.2022):

	```
	git clone https://github.com/lh3/bwa
	```
</details>
<br>

## BAM manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/samtools">samtools</a></b> (ğŸ¥‡25 Â·  â­ 1.5K Â· â•) - Tools (written in C using htslib) for manipulating next-generation sequencing data.</summary>

- [GitHub](https://github.com/samtools/samtools) (ğŸ‘¨â€ğŸ’» 92 Â· ğŸ”€ 540 Â· ğŸ“¥ 1.4M Â· ğŸ“‹ 1.2K - 14% open Â· â±ï¸ 26.01.2024):

	```
	git clone https://github.com/samtools/samtools
	```
</details>
<details><summary><b><a href="https://github.com/biod/sambamba">sambamba</a></b> (ğŸ¥ˆ20 Â·  â­ 540 Â· â•) - Tools for working with SAM/BAM data.</summary>

- [GitHub](https://github.com/biod/sambamba) (ğŸ‘¨â€ğŸ’» 29 Â· ğŸ”€ 96 Â· ğŸ“¥ 140K Â· ğŸ“‹ 450 - 6% open Â· â±ï¸ 21.08.2023):

	```
	git clone https://github.com/biod/sambamba
	```
</details>
<details><summary><b><a href="https://github.com/brentp/mosdepth">mosdepth</a></b> (ğŸ¥‰19 Â·  â­ 630 Â· â•) - fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing.</summary>

- [GitHub](https://github.com/brentp/mosdepth) (ğŸ‘¨â€ğŸ’» 16 Â· ğŸ”€ 99 Â· ğŸ“¥ 300K Â· ğŸ“‹ 200 - 24% open Â· â±ï¸ 22.11.2023):

	```
	git clone https://github.com/brentp/mosdepth
	```
</details>
<details><summary><b><a href="https://github.com/GregoryFaust/samblaster">samblaster</a></b> (ğŸ¥‰14 Â·  â­ 210 Â· ğŸ’€) - samblaster: a tool to mark duplicates and extract discordant and split reads from sam files.</summary>

- [GitHub](https://github.com/GregoryFaust/samblaster) (ğŸ‘¨â€ğŸ’» 5 Â· ğŸ”€ 29 Â· ğŸ“¥ 23K Â· ğŸ“‹ 49 - 26% open Â· â±ï¸ 04.06.2020):

	```
	git clone https://github.com/GregoryFaust/samblaster
	```
</details>
<br>

## Variant calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

