<!-- markdownlint-disable -->
<h1 align="center">
    best-of-NGS-Germline
    <br>
</h1>

## Contents

- [FASTQ quality control](#fastq-quality-control) _3 projects_
- [Read alignment](#read-alignment) _2 projects_
- [BAM manipulation and stats](#bam-manipulation-and-stats) _4 projects_
- [Variant calling](#variant-calling) _3 projects_
- [VCF manipulation and stats](#vcf-manipulation-and-stats) _2 projects_
- [Copy number variation calling](#copy-number-variation-calling) _1 projects_
- [Structural variation calling](#structural-variation-calling) _1 projects_
- [Variant annotation](#variant-annotation) _3 projects_
- [Gene typing](#gene-typing) _4 projects_
- [BED manipulation and stats](#bed-manipulation-and-stats) _3 projects_
- [Relationship inference](#relationship-inference) _1 projects_
- [FASTQ compression](#fastq-compression) _1 projects_

<br>

## FASTQ quality control

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/OpenGene/fastp">OpenGene/fastp</a></b> (ğŸ¥‰20 Â·  â­ 1.7K Â· ğŸ’¤) - An ultra-fast all-in-one FASTQ preprocessor (QC/adapters/trimming/filtering/splitting/merging...).</summary>

- [GitHub](https://github.com/OpenGene/fastp) (ğŸ‘¨â€ğŸ’» 26 Â· ğŸ”€ 320 Â· ğŸ“‹ 490 - 63% open Â· â±ï¸ 31.05.2023):

	```
	git clone https://github.com/OpenGene/fastp
	```
</details>
<details><summary><b><a href="https://github.com/marcelm/cutadapt">marcelm/cutadapt</a></b> (ğŸ¥‡25 Â·  â­ 480) - Cutadapt removes adapter sequences from sequencing reads.</summary>

- [GitHub](https://github.com/marcelm/cutadapt) (ğŸ‘¨â€ğŸ’» 30 Â· ğŸ”€ 130 Â· ğŸ“¥ 4.3K Â· ğŸ“¦ 400 Â· ğŸ“‹ 680 - 8% open Â· â±ï¸ 05.02.2024):

	```
	git clone https://github.com/marcelm/cutadapt
	```
</details>
<details><summary><b><a href="https://github.com/timflutre/trimmomatic">timflutre/trimmomatic</a></b> (ğŸ¥‰5 Â·  â­ 120 Â· ğŸ’€) - Read trimming tool for Illumina NGS data.</summary>

- [GitHub](https://github.com/timflutre/trimmomatic) (ğŸ‘¨â€ğŸ’» 2 Â· ğŸ”€ 88 Â· ğŸ“‹ 21 - 33% open Â· â±ï¸ 10.03.2015):

	```
	git clone https://github.com/timflutre/trimmomatic
	```
</details>
<br>

## Read alignment

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/lh3/bwa">lh3/bwa</a></b> (ğŸ¥‰19 Â·  â­ 1.4K Â· ğŸ’€) - Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment).</summary>

- [GitHub](https://github.com/lh3/bwa) (ğŸ‘¨â€ğŸ’» 33 Â· ğŸ”€ 510 Â· ğŸ“¥ 100K Â· ğŸ“‹ 280 - 63% open Â· â±ï¸ 22.09.2022):

	```
	git clone https://github.com/lh3/bwa
	```
</details>
<details><summary><b><a href="https://github.com/BenLangmead/bowtie2">BenLangmead/bowtie2</a></b> (ğŸ¥‡23 Â·  â­ 600) - A fast and sensitive gapped read aligner.</summary>

- [GitHub](https://github.com/BenLangmead/bowtie2) (ğŸ‘¨â€ğŸ’» 38 Â· ğŸ”€ 160 Â· ğŸ“¥ 71K Â· ğŸ“‹ 400 - 34% open Â· â±ï¸ 17.01.2024):

	```
	git clone https://github.com/BenLangmead/bowtie2
	```
</details>
<br>

## BAM manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/samtools">samtools/samtools</a></b> (ğŸ¥‡25 Â·  â­ 1.5K) - Tools (written in C using htslib) for manipulating next-generation sequencing data.</summary>

- [GitHub](https://github.com/samtools/samtools) (ğŸ‘¨â€ğŸ’» 93 Â· ğŸ”€ 550 Â· ğŸ“¥ 1.4M Â· ğŸ“‹ 1.2K - 14% open Â· â±ï¸ 07.02.2024):

	```
	git clone https://github.com/samtools/samtools
	```
</details>
<details><summary><b><a href="https://github.com/brentp/mosdepth">brentp/mosdepth</a></b> (ğŸ¥‰19 Â·  â­ 630) - fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing.</summary>

- [GitHub](https://github.com/brentp/mosdepth) (ğŸ‘¨â€ğŸ’» 16 Â· ğŸ”€ 99 Â· ğŸ“¥ 300K Â· ğŸ“‹ 200 - 24% open Â· â±ï¸ 22.11.2023):

	```
	git clone https://github.com/brentp/mosdepth
	```
</details>
<details><summary><b><a href="https://github.com/biod/sambamba">biod/sambamba</a></b> (ğŸ¥ˆ20 Â·  â­ 540) - Tools for working with SAM/BAM data.</summary>

- [GitHub](https://github.com/biod/sambamba) (ğŸ‘¨â€ğŸ’» 29 Â· ğŸ”€ 96 Â· ğŸ“¥ 140K Â· ğŸ“‹ 450 - 6% open Â· â±ï¸ 21.08.2023):

	```
	git clone https://github.com/biod/sambamba
	```
</details>
<details><summary><b><a href="https://github.com/GregoryFaust/samblaster">GregoryFaust/samblaster</a></b> (ğŸ¥‰14 Â·  â­ 210 Â· ğŸ’€) - samblaster: a tool to mark duplicates and extract discordant and split reads from sam files.</summary>

- [GitHub](https://github.com/GregoryFaust/samblaster) (ğŸ‘¨â€ğŸ’» 5 Â· ğŸ”€ 29 Â· ğŸ“¥ 23K Â· ğŸ“‹ 49 - 26% open Â· â±ï¸ 04.06.2020):

	```
	git clone https://github.com/GregoryFaust/samblaster
	```
</details>
<br>

## Variant calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/google/deepvariant">google/deepvariant</a></b> (ğŸ¥‰23 Â·  â­ 3K) - DeepVariant is an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA..</summary>

- [GitHub](https://github.com/google/deepvariant) (ğŸ‘¨â€ğŸ’» 28 Â· ğŸ”€ 680 Â· ğŸ“¥ 4.7K Â· ğŸ“‹ 710 - 0% open Â· â±ï¸ 09.12.2023):

	```
	git clone https://github.com/google/deepvariant
	```
</details>
<details><summary><b><a href="https://github.com/broadinstitute/gatk">broadinstitute/gatk</a></b> (ğŸ¥‡25 Â·  â­ 1.6K) - Official code repository for GATK versions 4 and up.</summary>

- [GitHub](https://github.com/broadinstitute/gatk) (ğŸ‘¨â€ğŸ’» 130 Â· ğŸ”€ 550 Â· ğŸ“¥ 560K Â· ğŸ“‹ 4.5K - 26% open Â· â±ï¸ 07.02.2024):

	```
	git clone https://github.com/broadinstitute/gatk
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/strelka">Illumina/strelka</a></b> (ğŸ¥‰18 Â·  â­ 340 Â· ğŸ’€) - Strelka2 germline and somatic small variant caller.</summary>

- [GitHub](https://github.com/Illumina/strelka) (ğŸ‘¨â€ğŸ’» 32 Â· ğŸ”€ 99 Â· ğŸ“¥ 34K Â· ğŸ“‹ 230 - 53% open Â· â±ï¸ 06.02.2019):

	```
	git clone https://github.com/Illumina/strelka
	```
</details>
<br>

## VCF manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/bcftools">samtools/bcftools</a></b> (ğŸ¥‡23 Â·  â­ 600) - This is the official development repository for BCFtools. See installation instructions and other documentation here..</summary>

- [GitHub](https://github.com/samtools/bcftools) (ğŸ‘¨â€ğŸ’» 54 Â· ğŸ”€ 220 Â· ğŸ“¥ 560K Â· ğŸ“‹ 1.8K - 15% open Â· â±ï¸ 08.02.2024):

	```
	git clone https://github.com/samtools/bcftools
	```
</details>
<details><summary><b><a href="https://github.com/vcftools/vcftools">vcftools/vcftools</a></b> (ğŸ¥‰16 Â·  â­ 460 Â· ğŸ’¤) - A set of tools written in Perl and C++ for working with VCF files, such as those generated by the 1000 Genomes Project.</summary>

- [GitHub](https://github.com/vcftools/vcftools) (ğŸ‘¨â€ğŸ’» 20 Â· ğŸ”€ 140 Â· ğŸ“¥ 110K Â· ğŸ“‹ 180 - 56% open Â· â±ï¸ 14.03.2023):

	```
	git clone https://github.com/vcftools/vcftools
	```
</details>
<br>

## Copy number variation calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/etal/cnvkit">etal/cnvkit</a></b> (ğŸ¥‡18 Â·  â­ 460) - Copy number variant detection from targeted DNA sequencing.</summary>

- [GitHub](https://github.com/etal/cnvkit) (ğŸ‘¨â€ğŸ’» 38 Â· ğŸ”€ 150 Â· ğŸ“¦ 28 Â· ğŸ“‹ 720 - 39% open Â· â±ï¸ 11.12.2023):

	```
	git clone https://github.com/etal/cnvkit
	```
</details>
<br>

## Structural variation calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/Illumina/manta">Illumina/manta</a></b> (ğŸ¥‡19 Â·  â­ 380 Â· ğŸ’€) - Structural variant and indel caller for mapped sequencing data.</summary>

- [GitHub](https://github.com/Illumina/manta) (ğŸ‘¨â€ğŸ’» 24 Â· ğŸ”€ 150 Â· ğŸ“¥ 42K Â· ğŸ“‹ 300 - 42% open Â· â±ï¸ 17.07.2019):

	```
	git clone https://github.com/Illumina/manta
	```
</details>
<br>

## Variant annotation

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/Ensembl/ensembl-vep">Ensembl/ensembl-vep</a></b> (ğŸ¥‡19 Â·  â­ 410) - The Ensembl Variant Effect Predictor predicts the functional effects of genomic variants.</summary>

- [GitHub](https://github.com/Ensembl/ensembl-vep) (ğŸ‘¨â€ğŸ’» 45 Â· ğŸ”€ 140 Â· ğŸ“‹ 980 - 9% open Â· â±ï¸ 17.01.2024):

	```
	git clone https://github.com/Ensembl/ensembl-vep
	```
</details>
<details><summary><b><a href="https://github.com/pcingola/SnpEff">pcingola/SnpEff</a></b> (ğŸ¥‰15 Â·  â­ 220) - </summary>

- [GitHub](https://github.com/pcingola/SnpEff) (ğŸ‘¨â€ğŸ’» 21 Â· ğŸ”€ 71 Â· ğŸ“‹ 460 - 3% open Â· â±ï¸ 17.01.2024):

	```
	git clone https://github.com/pcingola/SnpEff
	```
</details>
<details><summary><b><a href="https://github.com/WGLab/doc-ANNOVAR">WGLab/doc-ANNOVAR</a></b> (ğŸ¥‰12 Â·  â­ 200) - Documentation for the ANNOVAR software.</summary>

- [GitHub](https://github.com/WGLab/doc-ANNOVAR) (ğŸ‘¨â€ğŸ’» 8 Â· ğŸ”€ 290 Â· ğŸ“‹ 230 - 47% open Â· â±ï¸ 28.11.2023):

	```
	git clone https://github.com/WGLab/doc-ANNOVAR
	```
</details>
<br>

## Gene typing

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/FRED-2/OptiType">FRED-2/OptiType</a></b> (ğŸ¥ˆ13 Â·  â­ 170 Â· ğŸ’€) - Precision HLA typing from next-generation sequencing data.</summary>

- [GitHub](https://github.com/FRED-2/OptiType) (ğŸ‘¨â€ğŸ’» 13 Â· ğŸ”€ 69 Â· ğŸ“‹ 110 - 43% open Â· â±ï¸ 15.09.2020):

	```
	git clone https://github.com/FRED-2/OptiType
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/ExpansionHunter">Illumina/ExpansionHunter</a></b> (ğŸ¥‡14 Â·  â­ 160) - A tool for estimating repeat sizes.</summary>

- [GitHub](https://github.com/Illumina/ExpansionHunter) (ğŸ‘¨â€ğŸ’» 17 Â· ğŸ”€ 49 Â· ğŸ“¥ 9.7K Â· ğŸ“‹ 100 - 60% open Â· â±ï¸ 30.01.2024):

	```
	git clone https://github.com/Illumina/ExpansionHunter
	```
</details>
<details><summary><b><a href="https://github.com/humanlongevity/HLA">humanlongevity/HLA</a></b> (ğŸ¥‰8 Â·  â­ 94 Â· ğŸ’€) - xHLA: Fast and accurate HLA typing from short read sequence data.</summary>

- [GitHub](https://github.com/humanlongevity/HLA) (ğŸ‘¨â€ğŸ’» 4 Â· ğŸ”€ 46 Â· ğŸ“‹ 63 - 73% open Â· â±ï¸ 04.10.2017):

	```
	git clone https://github.com/humanlongevity/HLA
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/SMNCopyNumberCaller">Illumina/SMNCopyNumberCaller</a></b> (ğŸ¥‰8 Â·  â­ 44) - A copy number caller for SMN1 and SMN2 to enable SMA diagnosis and carrier screening with WGS.</summary>

- [GitHub](https://github.com/Illumina/SMNCopyNumberCaller) (ğŸ‘¨â€ğŸ’» 4 Â· ğŸ”€ 11 Â· ğŸ“‹ 10 - 50% open Â· â±ï¸ 14.10.2023):

	```
	git clone https://github.com/Illumina/SMNCopyNumberCaller
	```
</details>
<br>

## BED manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/arq5x/bedtools2">arq5x/bedtools2</a></b> (ğŸ¥‡24 Â·  â­ 880) - bedtools - the swiss army knife for genome arithmetic.</summary>

- [GitHub](https://github.com/arq5x/bedtools2) (ğŸ‘¨â€ğŸ’» 93 Â· ğŸ”€ 280 Â· ğŸ“¥ 970K Â· ğŸ“‹ 810 - 25% open Â· â±ï¸ 07.11.2023):

	```
	git clone https://github.com/arq5x/bedtools2
	```
</details>
<details><summary><b><a href="https://github.com/lh3/bedtk">lh3/bedtk</a></b> (ğŸ¥‰7 Â·  â­ 120 Â· ğŸ’€) - A simple toolset for BED files (warning: CLI may change before bedtk becomes stable).</summary>

- [GitHub](https://github.com/lh3/bedtk) (ğŸ‘¨â€ğŸ’» 3 Â· ğŸ”€ 13 Â· â±ï¸ 01.04.2022):

	```
	git clone https://github.com/lh3/bedtk
	```
</details>
<details><summary><b><a href="https://github.com/lh3/tabtk">lh3/tabtk</a></b> (ğŸ¥‰4 Â·  â­ 59 Â· ğŸ’€) - Toolkit for processing TAB-delimited format.</summary>

- [GitHub](https://github.com/lh3/tabtk) (ğŸ”€ 11 Â· ğŸ“‹ 2 - 50% open Â· â±ï¸ 11.08.2016):

	```
	git clone https://github.com/lh3/tabtk
	```
</details>
<br>

## Relationship inference

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/brentp/somalier">brentp/somalier</a></b> (ğŸ¥‡17 Â·  â­ 240) - fast sample-swap and relatedness checks on BAMs/CRAMs/VCFs/GVCFs... like damn that is one smart wine guy.</summary>

- [GitHub](https://github.com/brentp/somalier) (ğŸ‘¨â€ğŸ’» 8 Â· ğŸ”€ 31 Â· ğŸ“¥ 14K Â· ğŸ“‹ 120 - 39% open Â· â±ï¸ 02.02.2024):

	```
	git clone https://github.com/brentp/somalier
	```
</details>
<br>

## FASTQ compression

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/shubhamchandak94/Spring">shubhamchandak94/Spring</a></b> (ğŸ¥‡9 Â·  â­ 100 Â· ğŸ’¤) - FASTQ compression.</summary>

- [GitHub](https://github.com/shubhamchandak94/Spring) (ğŸ”€ 19 Â· â±ï¸ 27.05.2023):

	```
	git clone https://github.com/shubhamchandak94/Spring
	```
</details>
