<!-- markdownlint-disable -->
<h1 align="center">
    best-of-NGS-Germline
    <br>
</h1>

## Contents

- [FASTQ quality control](#fastq-quality-control) _3 projects_
- [Read alignment](#read-alignment) _2 projects_
- [BAM manipulation and stats](#bam-manipulation-and-stats) _4 projects_
- [Variant calling](#variant-calling) _3 projects_
- [VCF manipulation and stats](#vcf-manipulation-and-stats) _2 projects_
- [Copy number variation calling](#copy-number-variation-calling) _1 projects_
- [Structural variation calling](#structural-variation-calling) _1 projects_
- [Variant annotation](#variant-annotation) _3 projects_
- [Gene typing](#gene-typing) _4 projects_
- [BED manipulation and stats](#bed-manipulation-and-stats) _3 projects_
- [Relationship inference](#relationship-inference) _1 projects_
- [FASTQ compression](#fastq-compression) _1 projects_

<br>

## FASTQ quality control

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/OpenGene/fastp">OpenGene/fastp</a></b> (🥉20 ·  ⭐ 1.7K · 💤) - An ultra-fast all-in-one FASTQ preprocessor (QC/adapters/trimming/filtering/splitting/merging...).</summary>

- [GitHub](https://github.com/OpenGene/fastp) (👨‍💻 26 · 🔀 320 · 📋 490 - 63% open · ⏱️ 31.05.2023):

	```
	git clone https://github.com/OpenGene/fastp
	```
</details>
<details><summary><b><a href="https://github.com/marcelm/cutadapt">marcelm/cutadapt</a></b> (🥇25 ·  ⭐ 480) - Cutadapt removes adapter sequences from sequencing reads.</summary>

- [GitHub](https://github.com/marcelm/cutadapt) (👨‍💻 30 · 🔀 130 · 📥 4.3K · 📦 400 · 📋 680 - 8% open · ⏱️ 05.02.2024):

	```
	git clone https://github.com/marcelm/cutadapt
	```
</details>
<details><summary><b><a href="https://github.com/timflutre/trimmomatic">timflutre/trimmomatic</a></b> (🥉5 ·  ⭐ 120 · 💀) - Read trimming tool for Illumina NGS data.</summary>

- [GitHub](https://github.com/timflutre/trimmomatic) (👨‍💻 2 · 🔀 88 · 📋 21 - 33% open · ⏱️ 10.03.2015):

	```
	git clone https://github.com/timflutre/trimmomatic
	```
</details>
<br>

## Read alignment

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/lh3/bwa">lh3/bwa</a></b> (🥉19 ·  ⭐ 1.4K · 💀) - Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment).</summary>

- [GitHub](https://github.com/lh3/bwa) (👨‍💻 33 · 🔀 510 · 📥 100K · 📋 280 - 63% open · ⏱️ 22.09.2022):

	```
	git clone https://github.com/lh3/bwa
	```
</details>
<details><summary><b><a href="https://github.com/BenLangmead/bowtie2">BenLangmead/bowtie2</a></b> (🥇23 ·  ⭐ 600) - A fast and sensitive gapped read aligner.</summary>

- [GitHub](https://github.com/BenLangmead/bowtie2) (👨‍💻 38 · 🔀 160 · 📥 71K · 📋 400 - 34% open · ⏱️ 17.01.2024):

	```
	git clone https://github.com/BenLangmead/bowtie2
	```
</details>
<br>

## BAM manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/samtools">samtools/samtools</a></b> (🥇25 ·  ⭐ 1.5K) - Tools (written in C using htslib) for manipulating next-generation sequencing data.</summary>

- [GitHub](https://github.com/samtools/samtools) (👨‍💻 93 · 🔀 550 · 📥 1.4M · 📋 1.2K - 14% open · ⏱️ 07.02.2024):

	```
	git clone https://github.com/samtools/samtools
	```
</details>
<details><summary><b><a href="https://github.com/brentp/mosdepth">brentp/mosdepth</a></b> (🥉19 ·  ⭐ 630) - fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing.</summary>

- [GitHub](https://github.com/brentp/mosdepth) (👨‍💻 16 · 🔀 99 · 📥 300K · 📋 200 - 24% open · ⏱️ 22.11.2023):

	```
	git clone https://github.com/brentp/mosdepth
	```
</details>
<details><summary><b><a href="https://github.com/biod/sambamba">biod/sambamba</a></b> (🥈20 ·  ⭐ 540) - Tools for working with SAM/BAM data.</summary>

- [GitHub](https://github.com/biod/sambamba) (👨‍💻 29 · 🔀 96 · 📥 140K · 📋 450 - 6% open · ⏱️ 21.08.2023):

	```
	git clone https://github.com/biod/sambamba
	```
</details>
<details><summary><b><a href="https://github.com/GregoryFaust/samblaster">GregoryFaust/samblaster</a></b> (🥉14 ·  ⭐ 210 · 💀) - samblaster: a tool to mark duplicates and extract discordant and split reads from sam files.</summary>

- [GitHub](https://github.com/GregoryFaust/samblaster) (👨‍💻 5 · 🔀 29 · 📥 23K · 📋 49 - 26% open · ⏱️ 04.06.2020):

	```
	git clone https://github.com/GregoryFaust/samblaster
	```
</details>
<br>

## Variant calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/google/deepvariant">google/deepvariant</a></b> (🥉23 ·  ⭐ 3K) - DeepVariant is an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA..</summary>

- [GitHub](https://github.com/google/deepvariant) (👨‍💻 28 · 🔀 680 · 📥 4.7K · 📋 710 - 0% open · ⏱️ 09.12.2023):

	```
	git clone https://github.com/google/deepvariant
	```
</details>
<details><summary><b><a href="https://github.com/broadinstitute/gatk">broadinstitute/gatk</a></b> (🥇25 ·  ⭐ 1.6K) - Official code repository for GATK versions 4 and up.</summary>

- [GitHub](https://github.com/broadinstitute/gatk) (👨‍💻 130 · 🔀 550 · 📥 560K · 📋 4.5K - 26% open · ⏱️ 07.02.2024):

	```
	git clone https://github.com/broadinstitute/gatk
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/strelka">Illumina/strelka</a></b> (🥉18 ·  ⭐ 340 · 💀) - Strelka2 germline and somatic small variant caller.</summary>

- [GitHub](https://github.com/Illumina/strelka) (👨‍💻 32 · 🔀 99 · 📥 34K · 📋 230 - 53% open · ⏱️ 06.02.2019):

	```
	git clone https://github.com/Illumina/strelka
	```
</details>
<br>

## VCF manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/samtools/bcftools">samtools/bcftools</a></b> (🥇23 ·  ⭐ 600) - This is the official development repository for BCFtools. See installation instructions and other documentation here..</summary>

- [GitHub](https://github.com/samtools/bcftools) (👨‍💻 54 · 🔀 220 · 📥 560K · 📋 1.8K - 15% open · ⏱️ 08.02.2024):

	```
	git clone https://github.com/samtools/bcftools
	```
</details>
<details><summary><b><a href="https://github.com/vcftools/vcftools">vcftools/vcftools</a></b> (🥉16 ·  ⭐ 460 · 💤) - A set of tools written in Perl and C++ for working with VCF files, such as those generated by the 1000 Genomes Project.</summary>

- [GitHub](https://github.com/vcftools/vcftools) (👨‍💻 20 · 🔀 140 · 📥 110K · 📋 180 - 56% open · ⏱️ 14.03.2023):

	```
	git clone https://github.com/vcftools/vcftools
	```
</details>
<br>

## Copy number variation calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/etal/cnvkit">etal/cnvkit</a></b> (🥇18 ·  ⭐ 460) - Copy number variant detection from targeted DNA sequencing.</summary>

- [GitHub](https://github.com/etal/cnvkit) (👨‍💻 38 · 🔀 150 · 📦 28 · 📋 720 - 39% open · ⏱️ 11.12.2023):

	```
	git clone https://github.com/etal/cnvkit
	```
</details>
<br>

## Structural variation calling

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/Illumina/manta">Illumina/manta</a></b> (🥇19 ·  ⭐ 380 · 💀) - Structural variant and indel caller for mapped sequencing data.</summary>

- [GitHub](https://github.com/Illumina/manta) (👨‍💻 24 · 🔀 150 · 📥 42K · 📋 300 - 42% open · ⏱️ 17.07.2019):

	```
	git clone https://github.com/Illumina/manta
	```
</details>
<br>

## Variant annotation

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/Ensembl/ensembl-vep">Ensembl/ensembl-vep</a></b> (🥇19 ·  ⭐ 410) - The Ensembl Variant Effect Predictor predicts the functional effects of genomic variants.</summary>

- [GitHub](https://github.com/Ensembl/ensembl-vep) (👨‍💻 45 · 🔀 140 · 📋 980 - 9% open · ⏱️ 17.01.2024):

	```
	git clone https://github.com/Ensembl/ensembl-vep
	```
</details>
<details><summary><b><a href="https://github.com/pcingola/SnpEff">pcingola/SnpEff</a></b> (🥉15 ·  ⭐ 220) - </summary>

- [GitHub](https://github.com/pcingola/SnpEff) (👨‍💻 21 · 🔀 71 · 📋 460 - 3% open · ⏱️ 17.01.2024):

	```
	git clone https://github.com/pcingola/SnpEff
	```
</details>
<details><summary><b><a href="https://github.com/WGLab/doc-ANNOVAR">WGLab/doc-ANNOVAR</a></b> (🥉12 ·  ⭐ 200) - Documentation for the ANNOVAR software.</summary>

- [GitHub](https://github.com/WGLab/doc-ANNOVAR) (👨‍💻 8 · 🔀 290 · 📋 230 - 47% open · ⏱️ 28.11.2023):

	```
	git clone https://github.com/WGLab/doc-ANNOVAR
	```
</details>
<br>

## Gene typing

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/FRED-2/OptiType">FRED-2/OptiType</a></b> (🥈13 ·  ⭐ 170 · 💀) - Precision HLA typing from next-generation sequencing data.</summary>

- [GitHub](https://github.com/FRED-2/OptiType) (👨‍💻 13 · 🔀 69 · 📋 110 - 43% open · ⏱️ 15.09.2020):

	```
	git clone https://github.com/FRED-2/OptiType
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/ExpansionHunter">Illumina/ExpansionHunter</a></b> (🥇14 ·  ⭐ 160) - A tool for estimating repeat sizes.</summary>

- [GitHub](https://github.com/Illumina/ExpansionHunter) (👨‍💻 17 · 🔀 49 · 📥 9.7K · 📋 100 - 60% open · ⏱️ 30.01.2024):

	```
	git clone https://github.com/Illumina/ExpansionHunter
	```
</details>
<details><summary><b><a href="https://github.com/humanlongevity/HLA">humanlongevity/HLA</a></b> (🥉8 ·  ⭐ 94 · 💀) - xHLA: Fast and accurate HLA typing from short read sequence data.</summary>

- [GitHub](https://github.com/humanlongevity/HLA) (👨‍💻 4 · 🔀 46 · 📋 63 - 73% open · ⏱️ 04.10.2017):

	```
	git clone https://github.com/humanlongevity/HLA
	```
</details>
<details><summary><b><a href="https://github.com/Illumina/SMNCopyNumberCaller">Illumina/SMNCopyNumberCaller</a></b> (🥉8 ·  ⭐ 44) - A copy number caller for SMN1 and SMN2 to enable SMA diagnosis and carrier screening with WGS.</summary>

- [GitHub](https://github.com/Illumina/SMNCopyNumberCaller) (👨‍💻 4 · 🔀 11 · 📋 10 - 50% open · ⏱️ 14.10.2023):

	```
	git clone https://github.com/Illumina/SMNCopyNumberCaller
	```
</details>
<br>

## BED manipulation and stats

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/arq5x/bedtools2">arq5x/bedtools2</a></b> (🥇24 ·  ⭐ 880) - bedtools - the swiss army knife for genome arithmetic.</summary>

- [GitHub](https://github.com/arq5x/bedtools2) (👨‍💻 93 · 🔀 280 · 📥 970K · 📋 810 - 25% open · ⏱️ 07.11.2023):

	```
	git clone https://github.com/arq5x/bedtools2
	```
</details>
<details><summary><b><a href="https://github.com/lh3/bedtk">lh3/bedtk</a></b> (🥉7 ·  ⭐ 120 · 💀) - A simple toolset for BED files (warning: CLI may change before bedtk becomes stable).</summary>

- [GitHub](https://github.com/lh3/bedtk) (👨‍💻 3 · 🔀 13 · ⏱️ 01.04.2022):

	```
	git clone https://github.com/lh3/bedtk
	```
</details>
<details><summary><b><a href="https://github.com/lh3/tabtk">lh3/tabtk</a></b> (🥉4 ·  ⭐ 59 · 💀) - Toolkit for processing TAB-delimited format.</summary>

- [GitHub](https://github.com/lh3/tabtk) (🔀 11 · 📋 2 - 50% open · ⏱️ 11.08.2016):

	```
	git clone https://github.com/lh3/tabtk
	```
</details>
<br>

## Relationship inference

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/brentp/somalier">brentp/somalier</a></b> (🥇17 ·  ⭐ 240) - fast sample-swap and relatedness checks on BAMs/CRAMs/VCFs/GVCFs... like damn that is one smart wine guy.</summary>

- [GitHub](https://github.com/brentp/somalier) (👨‍💻 8 · 🔀 31 · 📥 14K · 📋 120 - 39% open · ⏱️ 02.02.2024):

	```
	git clone https://github.com/brentp/somalier
	```
</details>
<br>

## FASTQ compression

<a href="#contents"><img align="right" width="15" height="15" src="https://git.io/JtehR" alt="Back to top"></a>

<details><summary><b><a href="https://github.com/shubhamchandak94/Spring">shubhamchandak94/Spring</a></b> (🥇9 ·  ⭐ 100 · 💤) - FASTQ compression.</summary>

- [GitHub](https://github.com/shubhamchandak94/Spring) (🔀 19 · ⏱️ 27.05.2023):

	```
	git clone https://github.com/shubhamchandak94/Spring
	```
</details>
